### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by devYaoYH**
- Added Login Page

### CODE REVIEW

File: client/src/pages/Login.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            <Button variant="contained" color="primary" onClick={login}>
+              Log in
+            </Button>
+          </Grid>
+        </Grid>
+      )}
+      {(!isWeb || isDebug) && (
+        <Grid container spacing={5} direction="column" alignItems="center">
+          <Grid item>
+            <Typography variant="h3">$canAndG0</Typography>
```
Comment (cszqwe) [inline client/src/pages/Login.tsx:83]: Why this is $canAndG0?

File: client/src/pages/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  // Flag for us to manually run login on mobile
+  const [loginError, setLoginError] = useState(false);
+  // User Identity
+  const [identity, setIdentity] = useState<IdentityToken>(emptyIdentityToken());
+
+  const updateUser = (text: string) => {
+    const newIdentity = emptyIdentityToken();
+    newIdentity.sub = text;
+    setIdentity(newIdentity);
```
Comment (BhavyaLight) [inline client/src/pages/Login.tsx:22]:
```suggestion
    setIdentity(Object.assign({},newIdentity,{sub: text}));
```
Reason: preserves immutability. also wondering what does "sub" mean here?

File: client/src/pages/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  useEffect(() => {
+    if (identity.sub != "EMPTY" && identity.sub != "") {
+      window.location.href = "/home";
+    }
+  }, [identity]);
+
+  useEffect(() => {
+    // Initial login if on microapp
+    login();
+  }, []);
```
Comment (BhavyaLight) [inline client/src/pages/Login.tsx:49]: sorry if i have asked this before, is this also based on identity? if so do we want to add [identity] here?

File: client/src/pages/Login.tsx (hunk 1 above)
Comment (devYaoYH) [reply client/src/pages/Login.tsx]: resolved.

**Comment by devYaoYH** [comment]
Will merge pending adding of tests for Login page

File: client/src/pages/Login.tsx (hunk 3 above)
Comment (devYaoYH) [reply client/src/pages/Login.tsx:49]: answered in #74

File: client/src/pages/Login.tsx (hunk 2 above)
Comment (devYaoYH) [reply client/src/pages/Login.tsx:22]: Sub is a unique ID associated to each google account which is recommended for applications to use to identify users who logged in via their google account.

**Pull request merged**